File: python/telnet.py
import telnetlib, time

HOST = 'mxer.uwaterloo.ca'
tn = telnetlib.Telnet(HOST, 25)

tn.read_until('postmaster@uwaterloo.ca')
print 'read'

tn.write('HELO mxer.uwaterloo.ca\n')
tn.read_until('you')
print 'read'

tn.write('MAIL FROM: test@ing.com\n')
tn.read_until('ok')
print 'read'

tn.write('RCPT TO: a26sivakumaran@uwaterloo.ca\n')
tn.read_until('ok')
print 'read'

tn.write('DATA\n')
tn.read_until('itself')
print 'read'

tn.write('hi\n.\n')
tn.read_until('delivery')
print 'read'

tn.write('QUIT\n')
tn.read_until('connection')
print 'read'

tn.close()

File: telnet.js
const telnet = require('telnet-client');

const connection = new telnet();
const params = {host: 'mxer.uwaterloo.ca', port: 25, shellPrompt: /(\*+|postmaster@uwaterloo\.ca)/};

module.exports = (mailFrom, rcptTo, body) => {
  return connection.connect(params)
    .then((response) => {
      console.log(response);
    })
    .then((response) => {
      console.log(response);
      return connection.send('HELO mxer.uwaterloo.ca', {waitfor: 'you'});
    })
    .then((response) => {
      console.log(response);
      return connection.send(`MAIL FROM: ${mailFrom}`, {waitfor: 'ok'});
    })
    .then((response) => {
      console.log(response);
      return connection.send(`RCPT TO: ${rcptTo}`, {waitfor: 'ok'});
    })
    .then((response) => {
      console.log(response);
      return connection.send('DATA', {waitfor: 'itself'});
    })
    .then((response) => {
      console.log(response);
      return connection.send(`${body}\n.`, {waitfor: 'delivery'});
    })
    .then((response) => {
      console.log(response);
      return connection.send('QUIT', {waitfor: 'connection'});
    })
    .then((response) => {
      console.log(response);
      return connection.end();
    });
};
